### DISCUSSION TITLE
Propose to use Zulip as chat service for Spack

### DISCUSSION THREAD

**Discussion by fangohr** [spack/spack#33103] [General]
I want to suggest to the Spack team to use Zulip as a successor of the current Slack instance (https://slack.spack.io/) for chat-based discussions and support.

Reasons:

- Zulip provides at least the same functionality as Slack
  
- Zulip.com provides free hosting for open source projects without a message limit and without a maximum age of messages (https://zulip.com/for/open-source/)

- Zulip streams (corresponding to channels in Slack) can be made web-public: anybody can read (without needing an account), which may help use the discussions for user support (by making existing discussions indexable by search engines for example). Examples: https://rust-lang.zulipchat.com and https://chat.zulip.org/

- Messages in Zulip streams can have "topics" and are thus much better searchable in the future (https://zulip.com/help/streams-and-topics)

The most important point in my view is though that with the current Slack instance the discussions disappear after 90 days (https://slack.com/help/articles/115002422943-Message-file-and-app-limits-on-the-free-version-of-Slack)
 and thus waste effort from Spack developers and Spack contributors when solutions to problems are found and written down, but disappear after 90 days and then need to be re-found again when somebody else has the same problem.

I have evaluated a number of chat systems systematically and a short report is available at https://fangohr.github.io/blog/zulip.html if that is of interest. I can say - from my own experience - that the switch from Slack to Zulip made a huge difference for the remote and local teams affected. I could imagine this might work for the Spack team and community as well. 

Best wishes,

Hans

PS Rust is making use of Zulip (https://rust-lang.zulipchat.com), and there are claims Zulip makes a big difference to their progress (https://zulip.com/case-studies/rust/)

**Comment by haampie** [comment]
Hi @fangohr, incidentally I asked in the maintainers channel on Slack if we could move to something that's indexed by Google for Q&A, so that we don't need to answer the same questions over and over.

@tgamblin proposed to simply use Github Discussions, which makes sense to me. It's just that we should encourage people to ask questions here, since "discussions" doesn't sound like the place to get an answer to a basic question, meaning we need to promote it as such.

Regarding zulip: I've tried it twice and got annoyed with the ugly and non-intuitive interface, so I wouldn't find it very enjoyable... Also: it's difficult to move a community from one platform to another. If you drop slack, you'll lose a large fraction of people. If you don't close the slack workspace, you have two platforms without a guarantee that people will actually move.

And about Slack: the ephemeral nature of chat can also be an advantage since people don't have to worry a "stupid" question has their name on it forever

**Comment by fangohr** [comment]
Thanks @haampie .

In reverse order:

Regarding the "ephemeral" nature of Slack: of course that could be an advantage. I don't think it could have mattered hugely though as (i) you would need to know that this rule exits and (ii) it came only into force in July 2022. 

Regarding not being able to move the large user base on Slack currently to Zulip: I think it is important to look forward and occasionally make a hard cut to improve things in the long run. We have seen this with the Python2 to Python3 transition - very controversial at the time, now widely accepted as the right thing to do. I could also imagine that Zulip would automatically become the dominant forum (at least if in direct comparison to Slack) because the answers are indexed and show up in search engines (where as on Slack they disappear after 90 days, and during the 90 days you can only find them if you log in and search).

Regarding the user interface: While Zulip's interface is marginally more complex [because you have one additional layer of hierarchy due to being able to organise messages into 'topics' within a stream(=channel)] and might need getting use to, I don't think it is fundamentally ugly or hard to use. It is true that that people generally like what they know already, so if the core developers would not embrace a new technology, there may not be much point to consider this further.

----

Main reply: thank you for sharing @tgamblin's proposal to use the Github Discussions, which would address the main problem of information being lost due to the 90 day expiration time. I think this would be a significant improvement over the current situation.

----

As we discuss "discussions" as an alternative to chat-clients, here some additional thoughts on why a chat-based client (such as Zulip) might have additional advantages over a forum (such as github discussions):

(i) in addition to exchanging and archiving public messages, you can exchange private messages in the chat client: either directly one-to-one person, or in small ad-hoc groups. These are not archived, and not visible by anybody else but can be really useful as a middle ground between an email and video/audio-call. In particular for project developers to coordinate amongst each other in remote teams they can be very effective.

(ii) some users [and developers] find the more interactive nature of the 'chat', including the option to read and contribute from mobile phones etc, much more attractive than a conventional discussions forum.



Best wishes,

Hans
